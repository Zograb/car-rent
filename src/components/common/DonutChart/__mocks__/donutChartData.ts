const donutChartData = [
  {
    name: 'Part 1',
    value: 360,
    color: '#52C93F',
  },
  {
    name: 'Part 2',
    value: 230,
    color: '#006AFF',
  },
  {
    name: 'Part 3',
    value: 150,
    color: '#FF2727',
  },
  {
    name: 'Part 4',
    value: 230,
    color: '#FF2727',
  },
  {
    name: 'Part 5',
    value: 150,
    color: '#FFFF00',
  },
];

export default donutChartData;
