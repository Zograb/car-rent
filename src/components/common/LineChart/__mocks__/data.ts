export const lineChartData = [
  { label: 'Label 1', value1: 222, value2: 450 },
  { label: 'Label 2', value1: 120, value2: 250 },
  { label: 'Label 3', value1: 340, value2: 320 },
  { label: 'Label 4', value1: 190, value2: 400 },
  { label: 'Label 5', value1: 520, value2: 150 },
  { label: 'Label 6', value1: 320, value2: 370 },
  { label: 'Label 7', value1: 300, value2: 220 },
  { label: 'Label 8', value1: 600, value2: 200 },
  { label: 'Label 9', value1: 250, value2: 450 },
  { label: 'Label 10', value1: 450, value2: 250 },
];
