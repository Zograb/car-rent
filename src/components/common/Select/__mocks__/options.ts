const options = [
  { value: '6465', label: '6465' },
  { value: '5665', label: '5665' },
  { value: '1755', label: '1755' },
  { value: '4532', label: '4532' },
  { value: '5543', label: '5543' },
];

export default options;
