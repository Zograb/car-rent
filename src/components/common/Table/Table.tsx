import type { TableColumn, TableRow } from '@/types/Table';
import { type FC, memo } from 'react';
import 'twin.macro';

export interface TableProps {
  columns: TableColumn[];
  rows: TableRow[];
}

const Table: FC<TableProps> = memo(({ columns, rows }) => (
  <div tw="overflow-x-scroll lg:overflow-x-auto">
    <table tw="w-full min-w-table">
      <thead>
        <tr>
          {columns.map((column, index) => (
            <th
              key={column.key}
              align={index === columns.length - 1 ? 'right': 'left'}
              tw="font-light text-sm py-3"
            >
              {column.label}
            </th>
          ))}
        </tr>
      </thead>
      <tbody tw="[&>tr]:(border-t border-solid border-gray4)">
        {rows.map((row, index) => (
          <tr key={index}>
            {columns.map((column, index) => (
              <td
                key={column.key}
                align={index === columns.length - 1 ? 'right': 'left'}
                tw="py-4"
              >
                {row[column.key]}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  </div>
));

export default Table;
