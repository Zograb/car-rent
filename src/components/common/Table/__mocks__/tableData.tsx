import { FiTrash2 } from 'react-icons/fi';
import tw from 'twin.macro';

// Components
import { Button } from '@/components';

// Utils
import { formatNumberToCurrency } from '@/utils';

const row = {
  number: <div className="text-sm">01</div>,
  id: '12345',
  user: 'John Doe',
  status: 'Approved',
  earning: formatNumberToCurrency(3250),
  actions: (
    <div className="flex justify-end items-center">
      <Button size="small" className="mr-2 text-xl" icon={<FiTrash2 />} iconButton />
      <Button size="small">Details</Button>
    </div>
  ),
}

export const rows = [row, row, row, row];
export const columns = [
  { key: 'id', label: 'ID' },
  { key: 'user', label: 'User' },
  { key: 'status', label: 'Status' },
  { key: 'earning', label: 'Earning' },
  { key: 'actions', label: '' },
];
