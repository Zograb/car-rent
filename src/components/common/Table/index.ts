export { type TableProps, default as Table } from './Table';
