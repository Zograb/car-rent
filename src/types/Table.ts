import type { ReactNode } from 'react';

export interface TableColumn {
  key: string;
  label: string;
}

export interface TableRow {
  [key: string]: string | ReactNode;
}
