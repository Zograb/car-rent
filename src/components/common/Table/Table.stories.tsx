import type { Meta, StoryObj } from '@storybook/react';

import Table from './Table';
import { rows, columns } from './__mocks__/tableData';

const meta: Meta<typeof Table> = {
  title: 'Table',
  component: Table,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Table>;

export const Basic: Story = {
  args: {
    rows,
    columns,
  },
};
