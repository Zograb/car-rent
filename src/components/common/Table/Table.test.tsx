import { render } from '@testing-library/react';

import Table from './Table';
import { columns, rows } from './__mocks__/tableData';

describe('Select', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Table rows={rows} columns={columns} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render table with rows and columns', () => {
    const { getByRole } = render(
      <Table rows={rows} columns={columns} />
    );
    const table = getByRole('table');
    expect(table).toBeInTheDocument();
  });
});
