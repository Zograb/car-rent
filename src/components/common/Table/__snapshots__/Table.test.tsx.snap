// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
.c0 {
  overflow-x: scroll;
}

.c1 {
  width: 100%;
  min-width: 39rem;
}

.c2 {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 300;
}

.c3 >tr {
  border-top-width: 1px;
  border-style: solid;
  --tw-border-opacity: 1;
  border-color: rgb(222 222 222 / var(--tw-border-opacity));
}

.c4 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.c5 {
  border-radius: 0.25rem;
  font-weight: 500;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
  transition-duration: 200ms;
  transition-timing-function: linear;
  display: flex;
  align-items: center;
  justify-content: center;
  --tw-bg-opacity: 1;
  background-color: rgb(0 106 255 / var(--tw-bg-opacity));
  height: 2rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c5:hover {
  opacity: 0.8;
  transition-duration: 75ms;
}

.c5:active {
  opacity: 0.7;
  --tw-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1),0 1px 2px -1px rgb(0 0 0 / 0.1);
  --tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color),0 1px 2px -1px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow);
}

.c5:active {
  --tw-shadow-color: #006AFF;
  --tw-shadow: var(--tw-shadow-colored);
}

@media (min-width: 1024px) {
  .c0 {
    overflow-x: auto;
  }
}

<div>
  <div
    class="c0"
  >
    <table
      class="c1"
    >
      <thead>
        <tr>
          <th
            align="left"
            class="c2"
          >
            ID
          </th>
          <th
            align="left"
            class="c2"
          >
            User
          </th>
          <th
            align="left"
            class="c2"
          >
            Status
          </th>
          <th
            align="left"
            class="c2"
          >
            Earning
          </th>
          <th
            align="right"
            class="c2"
          />
        </tr>
      </thead>
      <tbody
        class="c3"
      >
        <tr>
          <td
            align="left"
            class="c4"
          >
            12345
          </td>
          <td
            align="left"
            class="c4"
          >
            John Doe
          </td>
          <td
            align="left"
            class="c4"
          >
            Approved
          </td>
          <td
            align="left"
            class="c4"
          >
            $3,250
          </td>
          <td
            align="right"
            class="c4"
          >
            <div
              class="flex justify-end items-center"
            >
              <button
                class="c5 mr-2 text-xl"
              >
                <span>
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="3 6 5 6 21 6"
                    />
                    <path
                      d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                    />
                    <line
                      x1="10"
                      x2="10"
                      y1="11"
                      y2="17"
                    />
                    <line
                      x1="14"
                      x2="14"
                      y1="11"
                      y2="17"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="c5"
              >
                <span />
                <span
                  class=""
                >
                  Details
                </span>
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            align="left"
            class="c4"
          >
            12345
          </td>
          <td
            align="left"
            class="c4"
          >
            John Doe
          </td>
          <td
            align="left"
            class="c4"
          >
            Approved
          </td>
          <td
            align="left"
            class="c4"
          >
            $3,250
          </td>
          <td
            align="right"
            class="c4"
          >
            <div
              class="flex justify-end items-center"
            >
              <button
                class="c5 mr-2 text-xl"
              >
                <span>
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="3 6 5 6 21 6"
                    />
                    <path
                      d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                    />
                    <line
                      x1="10"
                      x2="10"
                      y1="11"
                      y2="17"
                    />
                    <line
                      x1="14"
                      x2="14"
                      y1="11"
                      y2="17"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="c5"
              >
                <span />
                <span
                  class=""
                >
                  Details
                </span>
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            align="left"
            class="c4"
          >
            12345
          </td>
          <td
            align="left"
            class="c4"
          >
            John Doe
          </td>
          <td
            align="left"
            class="c4"
          >
            Approved
          </td>
          <td
            align="left"
            class="c4"
          >
            $3,250
          </td>
          <td
            align="right"
            class="c4"
          >
            <div
              class="flex justify-end items-center"
            >
              <button
                class="c5 mr-2 text-xl"
              >
                <span>
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="3 6 5 6 21 6"
                    />
                    <path
                      d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                    />
                    <line
                      x1="10"
                      x2="10"
                      y1="11"
                      y2="17"
                    />
                    <line
                      x1="14"
                      x2="14"
                      y1="11"
                      y2="17"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="c5"
              >
                <span />
                <span
                  class=""
                >
                  Details
                </span>
              </button>
            </div>
          </td>
        </tr>
        <tr>
          <td
            align="left"
            class="c4"
          >
            12345
          </td>
          <td
            align="left"
            class="c4"
          >
            John Doe
          </td>
          <td
            align="left"
            class="c4"
          >
            Approved
          </td>
          <td
            align="left"
            class="c4"
          >
            $3,250
          </td>
          <td
            align="right"
            class="c4"
          >
            <div
              class="flex justify-end items-center"
            >
              <button
                class="c5 mr-2 text-xl"
              >
                <span>
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="3 6 5 6 21 6"
                    />
                    <path
                      d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                    />
                    <line
                      x1="10"
                      x2="10"
                      y1="11"
                      y2="17"
                    />
                    <line
                      x1="14"
                      x2="14"
                      y1="11"
                      y2="17"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="c5"
              >
                <span />
                <span
                  class=""
                >
                  Details
                </span>
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
